runtime/src/set_auth_manifest.rs

```diff
@@ -401,6 +415,34 @@ impl SetAuthManifestCmd {
             soc_ifc,
         )?;
 
+        // Sort the image metadata list by firmware ID in place. Also check for duplicate firmware IDs.
+        let slice =
+            &mut metadata_mailbox.image_metadata_list[..metadata_mailbox.entry_count as usize];
```

Comment:
Why can't we use `metadata_mailbox.image_metadata_list.sort_unstable_by_key()`?

If we need to check for duplicates, we can do it after sorting.

Reply:
sort_unstable_by_key is increasing the Runtime size by 2+KB.

Reply:
in that case, we should add some tests for this, and we can make it more efficient later